### PR TITLE
MBLINTER-12 -profile screen UI and logout

### PR DESCRIPTION
**Pull request by marija-drobac**

### CODE REVIEW

File: app/src/main/java/com/example/hangoutz/ui/screens/settingsscreen/SettingsScreen.kt [hunk 1]
```diff
@@ -1,17 +1,177 @@
...
+        val (x, y) = createRefs()
+        Box(modifier = Modifier.constrainAs(x) {
+            start.linkTo(parent.start)
+            end.linkTo(parent.end)
+            top.linkTo(parent.top)
+            bottom.linkTo(parent.bottom)
+        }) {
+            nameField(
+                name,
+                { onNameChanged(it) },
```
Comment (marija-drobac) [inline app/src/main/java/com/example/hangoutz/ui/screens/settingsscreen/SettingsScreen.kt]: ddf

File: app/build.gradle [hunk 2]
```diff
@@ -68,6 +71,8 @@ dependencies {
     implementation libs.kotlinx.coroutines.core
     implementation(libs.hilt.android)
     implementation(libs.androidx.hilt.navigation.compose)
+    implementation libs.androidx.constraintlayout.compose
+    implementation  libs.compose
```
Comment (Mentors2024) [inline app/build.gradle]: Delete double space.

File: app/build.gradle [hunk 3]
```diff
@@ -68,6 +71,8 @@ dependencies {
     implementation libs.kotlinx.coroutines.core
     implementation(libs.hilt.android)
     implementation(libs.androidx.hilt.navigation.compose)
```
Comment (Mentors2024) [inline app/build.gradle]: Delete brackets.

File: app/src/main/java/com/example/hangoutz/di/Module.kt
Comment (Mentors2024) [inline app/src/main/java/com/example/hangoutz/di/Module.kt:1]: Format file.

File: app/src/main/java/com/example/hangoutz/domain/repository/UserRepository.kt
Comment (Mentors2024) [inline app/src/main/java/com/example/hangoutz/domain/repository/UserRepository.kt:1]: Format file.

File: app/src/main/java/com/example/hangoutz/ui/components/NameField.kt
Comment (Mentors2024) [inline app/src/main/java/com/example/hangoutz/ui/components/NameField.kt:1]: Format file.

File: app/src/main/java/com/example/hangoutz/ui/components/NameField.kt [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    modifier: Modifier
+
+) {
+        //val keyboart = remember { Keyboard }
+    BasicTextField(
+        value  = text,
+        onValueChange = onTextChange,
+        textStyle = TextStyle(
+            color = Ivory,
+            fontSize = 32.sp,
```
Comment (Mentors2024) [inline app/src/main/java/com/example/hangoutz/ui/components/NameField.kt]: Move font size to Type file.

File: app/src/main/java/com/example/hangoutz/ui/components/NameField.kt [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+            textAlign = TextAlign.Center,
+            ),
+        readOnly = isReadOnly,
+        singleLine = true,
+        modifier = modifier.fillMaxWidth()
+            .focusRequester(focusRequester),
+
```
Comment (Mentors2024) [inline app/src/main/java/com/example/hangoutz/ui/components/NameField.kt]: Delete empty lines.

File: app/src/main/java/com/example/hangoutz/ui/components/NameField.kt [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+@Composable
+fun nameField(
+    text: String,
+    onTextChange: (String) -> Unit,
+    isReadOnly: Boolean ,
+    focusRequester: FocusRequester,
+    modifier: Modifier
+
+) {
+        //val keyboart = remember { Keyboard }
```
Comment (Mentors2024) [inline app/src/main/java/com/example/hangoutz/ui/components/NameField.kt]: Delete unused line.

File: app/src/main/java/com/example/hangoutz/ui/components/NameField.kt
Comment (Mentors2024) [inline app/src/main/java/com/example/hangoutz/ui/components/NameField.kt:1]: Format file.

File: app/src/main/java/com/example/hangoutz/ui/screens/settingsscreen/SettingsScreen.kt
Comment (Mentors2024) [inline app/src/main/java/com/example/hangoutz/ui/screens/settingsscreen/SettingsScreen.kt:1]: Format file.

File: app/src/main/java/com/example/hangoutz/ui/screens/settingsscreen/SettingsScreen.kt [hunk 7]
```diff
@@ -1,17 +1,197 @@
...
+            Box(
+                modifier = Modifier
+                    .size(215.dp)
+                    .clip(CircleShape)
+                    .border(2.5f.dp, Ivory, CircleShape)
+            ) {
+            }
+            GlideImage(
+                model = "${BuildConfig.BASE_URL_AVATAR}${data.value.avatar}",
+                contentDescription = "",
```
Comment (Mentors2024) [inline app/src/main/java/com/example/hangoutz/ui/screens/settingsscreen/SettingsScreen.kt]: Is contentDescription optional if it is delete this line.

File: app/src/main/java/com/example/hangoutz/ui/screens/settingsscreen/SettingsScreen.kt [hunk 8]
```diff
@@ -1,17 +1,197 @@
...
+                    data.value.isReadOnly,
+                    { viewmodel.onPencilClick() },
+                    data.value.name
+                )
+            }
+            Text(
+                text = data.value.email,
+                style = MaterialTheme.typography.bodyLarge,
+                color = Ivory,
+                fontSize = 20.sp,
```
Comment (Mentors2024) [inline app/src/main/java/com/example/hangoutz/ui/screens/settingsscreen/SettingsScreen.kt]: Move to Type file.

File: app/src/main/java/com/example/hangoutz/ui/screens/settingsscreen/SettingsScreen.kt [hunk 9]
```diff
@@ -1,17 +1,197 @@
...
+                            navController.navigate(NavigationItem.Login.route)
+                        }
+                    )
+                }
+            )
+        }
+    }
+}
+
+@Composable
```
Comment (Mentors2024) [inline app/src/main/java/com/example/hangoutz/ui/screens/settingsscreen/SettingsScreen.kt]: Move to separate file.

File: app/src/main/java/com/example/hangoutz/ui/screens/settingsscreen/SettingsScreen.kt [hunk 10]
```diff
@@ -1,17 +1,197 @@
...
+fun NameInput(
```
Comment (Mentors2024) [inline app/src/main/java/com/example/hangoutz/ui/screens/settingsscreen/SettingsScreen.kt]: Its convention to pass variables first and then functions.

File: app/src/main/java/com/example/hangoutz/ui/screens/settingsscreen/SettingsScreen.kt [hunk 11]
```diff
@@ -1,17 +1,197 @@
...
+    onPencilClick: () -> Unit,
+    name: String
+) {
+    val focusRequester = remember { FocusRequester() }
+    ConstraintLayout(
+        modifier = Modifier
+            .fillMaxWidth()
+            .padding(end = 50.dp, start = 50.dp),
+    ) {
+        val (x, y) = createRefs()
```
Comment (Mentors2024) [inline app/src/main/java/com/example/hangoutz/ui/screens/settingsscreen/SettingsScreen.kt]: Rename x and y.

File: app/src/main/java/com/example/hangoutz/ui/screens/settingsscreen/SettingsScreen.kt [hunk 12]
```diff
@@ -1,17 +1,197 @@
...
+            top.linkTo(parent.top)
+            bottom.linkTo(parent.bottom)
+        }) {
+            nameField(
+                name,
+                { onNameChanged(it) },
+                isReadOnly,
+                focusRequester,
+                modifier = Modifier.align(Alignment.Center)
+
```
Comment (Mentors2024) [inline app/src/main/java/com/example/hangoutz/ui/screens/settingsscreen/SettingsScreen.kt]: Remove empty line.

File: app/src/main/java/com/example/hangoutz/ui/screens/settingsscreen/SettingsScreenViewModel.kt [hunk 13]
```diff
@@ -0,0 +1,78 @@
...
+    }
+
+    fun onNameChanged(newText: String) {
+        _uiState.value = _uiState.value.copy(name = newText)
+    }
+
+    fun onPencilClick() : Boolean{
+     val state =  !_uiState.value.isReadOnly
+        _uiState.value = _uiState.value.copy(isReadOnly = state)
+        //Log.e("LOGG------ ","STATE = ${state} & ISREADONLY ${_uiState.value.isReadOnly}")
```
Comment (Mentors2024) [inline app/src/main/java/com/example/hangoutz/ui/screens/settingsscreen/SettingsScreenViewModel.kt]: Remove commented line.

File: app/src/main/java/com/example/hangoutz/ui/screens/settingsscreen/SettingsScreenViewModel.kt [hunk 14]
```diff
@@ -0,0 +1,78 @@
...
+
```
Comment (Mentors2024) [inline app/src/main/java/com/example/hangoutz/ui/screens/settingsscreen/SettingsScreenViewModel.kt]: Remove empty line.

File: app/src/main/java/com/example/hangoutz/ui/screens/settingsscreen/SettingsScreenViewModel.kt [hunk 15]
```diff
@@ -0,0 +1,78 @@
...
+    }
+
+    fun getUser(context: Context) {
+
+        val userID = SharedPreferencesManager.getUserId(context)
+        viewModelScope.launch {
+            try {
+                val response =
+                    userRepository.getUserById(
+                        userID!!,
```
Comment (Mentors2024) [inline app/src/main/java/com/example/hangoutz/ui/screens/settingsscreen/SettingsScreenViewModel.kt]: Remove !!. If its not absolutely necessary don't use !!.

File: app/src/main/java/com/example/hangoutz/ui/screens/settingsscreen/SettingsScreen.kt
Comment (Mentors2024) [inline app/src/main/java/com/example/hangoutz/ui/screens/settingsscreen/SettingsScreen.kt:1]: Move all dp to Dimensions file. Read Srdjan message on skype. Do this anywhere in code where you have dimensions.

File: app/src/main/java/com/example/hangoutz/ui/screens/settingsscreen/SettingsScreenViewModel.kt [hunk 16]
```diff
@@ -0,0 +1,78 @@
...
+    init {
+        getUser(context)
+    }
+
+    fun onNameChanged(newText: String) {
+        _uiState.value = _uiState.value.copy(name = newText)
+    }
+
+    fun onPencilClick() : Boolean{
+     val state =  !_uiState.value.isReadOnly
```
Comment (Mentors2024) [inline app/src/main/java/com/example/hangoutz/ui/screens/settingsscreen/SettingsScreenViewModel.kt]: Rename this val state.

File: app/src/main/java/com/example/hangoutz/ui/screens/settingsscreen/SettingsScreenViewModel.kt [hunk 17]
```diff
@@ -0,0 +1,78 @@
...
+                    )
+                if (response.isSuccessful && !response.body().isNullOrEmpty()
+                ) {
+                    val user = response.body()?.first()
+                    user?.let {
+
+                        _uiState.value = _uiState.value.copy(
+                            name = it.name,
+                            email = it.email,
+                            avatar = if (it.avatar == null) "avatar_default.png" else it.avatar
```
Comment (Mentors2024) [inline app/src/main/java/com/example/hangoutz/ui/screens/settingsscreen/SettingsScreenViewModel.kt]: Move default logo to constants.

File: app/src/main/java/com/example/hangoutz/ui/screens/settingsscreen/SettingsScreenViewModel.kt
Comment (Mentors2024) [inline app/src/main/java/com/example/hangoutz/ui/screens/settingsscreen/SettingsScreenViewModel.kt:1]: Format file and remove empty lines.

**Review by Mentors2024 — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/hangoutz/ui/screens/settingsscreen/SettingsScreenViewModel.kt (hunk 15 above)
Comment (marija-drobac) [reply app/src/main/java/com/example/hangoutz/ui/screens/settingsscreen/SettingsScreenViewModel.kt]: It's necessary,  code doesn't work without it,

File: app/src/main/java/com/example/hangoutz/ui/screens/settingsscreen/NameInput.kt [hunk 18]
```diff
@@ -0,0 +1,74 @@
...
+            top.linkTo(parent.top)
+            bottom.linkTo(parent.bottom)
+        }) {
+            nameField(
+                name,
+                { onNameChanged(it) },
+                isReadOnly,
+                focusRequester,
+                modifier = Modifier.align(Alignment.Center)
+                    .testTag("nameField")
```
Comment (Mentors2024) [inline app/src/main/java/com/example/hangoutz/ui/screens/settingsscreen/NameInput.kt]: Move to Constants.

Comment (Mentors2024) [reply app/src/main/java/com/example/hangoutz/ui/screens/settingsscreen/NameInput.kt]: Do this for all test tags.

File: app/src/main/java/com/example/hangoutz/ui/components/NameField.kt [hunk 19]
```diff
@@ -0,0 +1,33 @@
...
+import com.example.hangoutz.utils.Constants.SETTINGS_NAME_FIELD_TAG
+
+@Composable
+fun nameField(
+    text: String,
+    onTextChange: (String) -> Unit,
+    isReadOnly: Boolean,
+    focusRequester: FocusRequester,
+    modifier: Modifier
+
```
Comment (Mentors2024) [inline app/src/main/java/com/example/hangoutz/ui/components/NameField.kt:20]: Remove emtpy line.

**Review by Mentors2024 — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/hangoutz/ui/screens/login/LoginScreen.kt [hunk 20]
```diff
@@ -119,16 +126,18 @@ fun LoginScreen(navController: NavController, viewmodel: LoginViewModel = hiltVi
                 text = getString(context, R.string.or_text),
                 color = Ivory,
                 modifier = Modifier,
+                style = MaterialTheme.typography.labelLarge,
                 textAlign = TextAlign.Center
             )
             Text(
                 text = getString(context, R.string.create_account_text),
```
Comment (Mentors2024) [inline app/src/main/java/com/example/hangoutz/ui/screens/login/LoginScreen.kt]: stringResources(R.string.creat.e...)

File: app/src/main/java/com/example/hangoutz/ui/screens/settings/SettingsScreenViewModel.kt [hunk 21]
```diff
@@ -0,0 +1,77 @@
...
+class SettingsViewModel @Inject constructor(
+    private val userRepository: UserRepository,
+    private val context: Context
+) : ViewModel() {
+
+    private val _uiState = MutableStateFlow(SettingsData())
+    val uiState: StateFlow<SettingsData> = _uiState
+
+    init {
+        getUser(context)
```
Comment (Mentors2024) [inline app/src/main/java/com/example/hangoutz/ui/screens/settings/SettingsScreenViewModel.kt]: No need to pass context, it is injected in viewModel.

**Review by Mentors2024 — CHANGES_REQUESTED** [review]

**Review by Mentors2024 — APPROVED** [review]

**Pull request merged**